src/charm.py

```diff
@@ -293,8 +292,16 @@ def _get_redis_relation_data(self) -> typing.Tuple[str, int]:
         for redis_unit in self._stored.redis_relation:  # type: ignore
             # mypy fails to see that this is indexable
             redis_unit_data = self._stored.redis_relation[redis_unit]  # type: ignore
-            redis_hostname = redis_unit_data.get("hostname", "")  # type: ignore
-            redis_port = redis_unit_data.get("port", DEFAULT_REDIS_PORT)  # type: ignore
+            try:
+                redis_hostname = str(redis_unit_data.get("hostname"))  # type: ignore
+            # I need to catch all exceptions that str() can throw
+            except Exception:  # pylint: disable=broad-exception-caught
+                redis_hostname = ""
+            try:
+                redis_port = int(redis_unit_data.get("port"))  # type: ignore
+            # I need to catch all exceptions that int() can throw
+            except Exception:  # pylint: disable=broad-exception-caught
```

Comment:
I think `int` will only throw `TypeError` or `ValueError` since the relation data are always string typed, so the value of `redis_unit_data.get("port")` will either be a string, or `None`.

Reply:
Actually I have a new idea, we can
```python
redis_port_str = redis_unit_data.get("port", "")
redis_port = int(redis_port_str) if redis_port_str.isnumeric() else 0
```

Reply:
Sorry I was wrong about this, we also need to consider the situation of `redis_unit_data.get("port", "")` being `None`.

Reply:
Outdated

Reply:
```python3
redis_port_data = redis_unit_data.get("port")
redis_port = int(redis_port_str) if redis_port_data and redis_port_data.isnumeric() else 0
```